### ISSUE TITLE
Register player

### ISSUE THREAD

**Issue by dragonjet** [KC3Kai/kc3-website#1]
API calls to be able to register a player

Accepts:
* Game Server
* Game ID
* Player name

Registers the user onto OUR database using `1SSGGGGGGGG` where `S` is the server number and `G` is the player game ID.

Player columns
* ID
* server
* game ID
* username (playername removed spaces)
* avatar link (gravatar or something, as long as it's a link, NOT upload)
* join date
* update date

**Comment by bamboo3250** [comment]
For this feature, I imagine we will have a login/register button at KC3Kai panel for redirecting the user to the website and submitting their information. Is that correct?

**Comment by dragonjet** [comment]
yes, on the planned "steam screen" by pressing <kbd>Tab</kbd>, if the user is not yet connected, there will be a probably "Join the masterrace" button :laughing:

Hopefully the registration form would also be on the extension, not the website - that is IF we still need a form. I'm thinking just use his ID and in-game admiral name. though I'm not sure if admiral names are re-usable across servers

**Comment by bamboo3250** [comment]
I think we should use the in-game admiral name as a display name only, not the one used to login or to identify the user. It may not be unique across servers as well though. But i'm more concerned about the security. Someone may play prank with us.
Anyway, I guess now it's time to visit the `dev-social` branch again xD

**Comment by dragonjet** [comment]
welp the only thing I've done on that branch is the <kbd>Tab</kbd> overlay, nothing else :laughing: 

but yea security is an issue with cross-access. so one thing we may need is a password at least.

but for making artificial calls to our server, like making it look like I got a Prinz while I didn't, shouldn't be our problem. we probably can't do anything about it being open source and all, we're easily reverse-engineerable :sweat:

~~though we could probably use a salt (encryption salt, not event salt) that is only saved on my side, or generated upon webstore builds, but JS being JS, they could inspect their local webstore copies and see the salt for ecryption, but nvm i don't know anyone who would go through these lengths just to cheat us or their friends :laughing: ~~

**Comment by Crazyteitoku** [comment]
Make use of One Time Pad (OTP) like how Github is setup. ![](http://vignette4.wikia.nocookie.net/kancolle/images/1/18/Pacman.png/revision/latest?cb=20150329071318)

**Comment by dragonjet** [comment]
why am I not getting github notifs for this, but getting emails

whats an OTP, welp have to google this some day

**Comment by Crazyteitoku** [comment]
[One-time Pad (wikipedia link)](https://en.wikipedia.org/wiki/One-time_pad)

**Comment by ethrundr** [comment]
I failed to understand....

so, basically, this is to register to *our* website?

**Comment by dragonjet** [comment]
yep, to our social system. remember this

![cqjt3noucaefhzs](https://cloud.githubusercontent.com/assets/557312/11530124/e74042b6-992c-11e5-84c7-d5c91c9f0619.jpg)
